CompilerSource/compiler/components/write_defragged_events.cpp

```diff
@@ -37,14 +38,49 @@ using namespace std;
 #include "event_reader/event_parser.h"
 #include "languages/lang_CPP.h"
 
-struct foundevent {
-  int mid, id, count;
-  foundevent(): mid(0),id(0),count(0) {}
-  void f2(int m, int i)  { id = i, mid = m; }
-  void inc(int m, int i) { mid = m, id = i, count++; }
-  void operator++(int) { count++; }
+struct EventGroup {
+  const EventDescriptor *base_event = nullptr;
+  int count = 0;
+  EventGroup &operator+=(const Event *ev) {
+    if (!base_event) {
+      base_event = ev;
+    } else if (base_event != ev) {
+      if (base_event->internal_id != ev->internal_id) {
+        std::cerr << "Two distinct events seem to have the same name ("
+                  << base_event->BaseFunctionName() << ")! Check IDs "
+                  << base_event->internal_id << " (" << base_event->HumanName()
+                  << ") and " << ev->internal_id << " (" << ev->HumanName()
+                  << ")" << std::endl;
+      } else {
+        std::cerr << "FYI: two distinct event pointers exist for event "
+                  << base_event->internal_id << " (" << base_event->HumanName()
+                  << ")..." << std::endl;
+      }
+    }
+    return *this;
+  }
+  EventGroup &operator=(const EventDescriptor *ev) {
+    base_event = ev;
+    return *this;
+  }
+  const EventDescriptor *operator->() const { return base_event; }
+  EventGroup(const Event *e): base_event(e) {}
+  EventGroup() = default;
+};
+
+struct UsedEventIndex {
+  map<string, EventDescriptor> base_methods;
+  set<Event> all;
+  void insert(int mid, int id) {
```

Comment:
bertter names or a note here on what "mid" is versus "id" wouldn't hurt

Reply:
Being a bit picky fundies, mid is main id. But documentation never hurts, it can only help.